minetest/minetest
# Adjust PSM distortion to use entire SM texture

This PR aims to improve quality and visible distance of the dynamic shadows.

What's done:
- [x] Passing camera/player position to all the relevant shaders
- [x] New distortion function that is tied to the position of the player
- [x] Adjusted light frustum to render more area in front of the player
- [x] Adjusted filter radius to get more crisp shadows.

## To do

This PR is Ready for Review.

## How to test

1. Enable dynamic shadows and choose one of the presets
2. Start a game that supports shadows (devtest)
3. Shadows should look better than earlier
4. Performance should be the same or better

For example, Medium preset should have decent shadow quality and support viewing distance up to 200 nodes.
